Rename unique feature to globalsequence.

baldersheim opened the pull request:
Invert so that the first doc from the first node will sort first,
followed by the first doc from the second node, and the the last doc of the last node at the end.
It will produce number in the range (1 << 48) down to zero.
Sequence = (1 << 48) - ((docId << 16) | disributionKey)

@havardpe and @geirst PR

baldersheim committed:
Rename unique feature to globalsequence.
Invert so that the first doc from the first node will sort first,
followed by the first doc from the second node, and the the last doc of the last node at the end.
It will produce number in the range (1 << 48) down to zero.
Sequence = (1 << 48) - ((docId << 16) | disributionKey)

havardpe commented on searchlib/src/vespa/searchlib/features/global_sequence_feature.cpp:
```diff
@@ -0,0 +1,64 @@
+GlobalSequenceBlueprint::setup(const IIndexEnvironment & env, const ParameterList & )
+{
+    _distributionKey = env.getDistributionKey();
+    assert( _distributionKey < 0x80000);
```
what about the 3 bits we allow to overlap between distribution key and docid?

baldersheim replied:
Ouch, thanks. That should be 0x10000

havardpe commented on searchlib/src/vespa/searchlib/features/global_sequence_feature.cpp:
```diff
@@ -0,0 +1,64 @@
+    }
+
+    void execute(uint32_t docId) override {
+        outputs().set_number(0, ((1ul << 48u) - ((uint64_t(docId) << 16u) | _distributionKey)));
```
consider moving the magic to an inlined function in the header that can also be called by the test

baldersheim replied:
Fixed

havardpe commented on searchlib/src/vespa/searchlib/features/global_sequence_feature.cpp:
```diff
@@ -0,0 +1,64 @@
+}
+
+GlobalSequenceBlueprint::GlobalSequenceBlueprint() :
+    Blueprint("globalsequence"),
```
consider camel case to match naming of other features

geirst replied:
Yes, the name should use came case.

baldersheim replied:
globalsequence -> globalSequence

baldersheim committed:
Followup from PR comments.
- CamelCase globalsequence -> globalSequence
- 0x80000 -> 0x10000
- Factor out computation to header file and use in both test and globalSequence feature.
